### PR TITLE
build: :rocket: Remove Autotools and MSVC build systems

### PR DESCRIPTION
**Pull request by hebasto**
The ultimate goal :)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,150 +1,8 @@
...
-bitcoin-qt_test
-
-# Resources cpp
-qrc_*.cpp
-
-# Mac specific
-.DS_Store
-build
+/build*
+!/build-aux
```
Comment (vasild) [inline .gitignore:2]: Why unignore `/build-aux`?

File: doc/.gitignore [hunk 2]
```diff
@@ -1 +0,0 @@
-Doxyfile
```
Comment (vasild) [inline doc/.gitignore:1]: How do we generate `Doxyfile` with `cmake`? It only needs to replace the version:

```diff
--- doc/Doxyfile.in     2022-12-12 11:24:16.156991000 +0100
+++ doc/Doxyfile        2024-04-24 14:24:56.529418000 +0200
@@ -38,7 +38,7 @@
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = @PACKAGE_VERSION@
+PROJECT_NUMBER         = 27.99.0
 
 # Using the PROJECT_BRIEF tag one can provide an optional one line description
```

**Review by vasild — APPROVED** [review]
ACK 51706e0fa06659fff0c1eacd1c9d33ff97e3ec53

I guess if something is forgotten it can be removed later and it looks like nothing excessive is removed either (but `doc/Doxygen` file generation?)

File: doc/.gitignore (hunk 2 above)
Comment (hebasto) [reply doc/.gitignore:1]: @vasild 
> How do we generate `Doxyfile` with `cmake`?

https://github.com/hebasto/bitcoin/pull/165

File: .gitignore (hunk 1 above)
Comment (hebasto) [reply .gitignore:2]: To ensure that Autotools legacy won't sneak into CMake realm.

**Comment by hebasto** [comment]
Unit tests create an artifact in the source tree:
```
share/rpcauth/__pycache__/rpcauth.cpython-312.pyc
```

**Comment by hebasto** [comment]
> Unit tests create an artifact in the source tree:
> 
> ```
> share/rpcauth/__pycache__/rpcauth.cpython-312.pyc
> ```

Fixed.

**Review by vasild — APPROVED** [review]
ACK 7752690763668eea32759ac3132ff44e3269c664

**Comment by hebasto** [comment]
Rebased. More stuff have been removed from the `src/univalue` .

**Review by pablomartin4btc — COMMENTED** [review]
ACK 44296a2177cee44d6f23a8a279b273de8cd66f12

**Review by vasild — APPROVED** [review]
ACK 44296a2177cee44d6f23a8a279b273de8cd66f12

**Comment by vasild** [comment]
ACK 045f267577abc0f2788311d34d1c31d4b6493ffb

**Review by vasild — APPROVED** [review]
ACK 3e6cf6080691dea5a0270b9a0113e42d52c39c29

Looks good, also the two new commits since my last review.

**Comment by vasild** [comment]
Why restore `src/minisketch/.gitignore` and why drop commit 3e6cf60806 `test: Do not write Python bytecode to source directory`?

**Comment by hebasto** [comment]
> Why restore `src/minisketch/.gitignore`..?

The `src/minisketch` is a subtree and shouldn't be modified in this PR.

> and why drop commit [3e6cf60](https://github.com/hebasto/bitcoin/commit/3e6cf6080691dea5a0270b9a0113e42d52c39c29) `test: Do not write Python bytecode to source directory`?

It is still not clear which way is preferred. For more details, please refer to https://github.com/bitcoin/bitcoin/pull/30533.

**Review by vasild — APPROVED** [review]
ACK d3f1c8ac5e7df3025d8e21707054b4c32adfcdd4

**Comment by hebasto** [comment]
Rebased.

**Review by vasild — APPROVED** [review]
ACK ad12ccc5d1313e81e7cee397f1dc887e9d3f4f7e

**Review by pablomartin4btc — APPROVED** [review]
ACK ad12ccc5d1313e81e7cee397f1dc887e9d3f4f7e

Tested it building `depends` with `make` and `cmake` building using the toolchain.

**Comment by hebasto** [comment]
> Looks like this is now [bitcoin#30731](https://github.com/bitcoin/bitcoin/pull/30731) and [bitcoin#30664](https://github.com/bitcoin/bitcoin/pull/30664), so reviewers of this change may want to go there.

Closing this PR.

**Pull request closed without merging**